### PR TITLE
Onboard

### PR DESCRIPTION
**Pull request by nehayward**
Added my initial notes for on-boarding to help with future hires and to document some workflow methods.

### CODE REVIEW

File: onboard/onboard.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Prerequisites
+- Install following apps to setup dev machine:
+    - git
```
Comment (zpatrick) [inline onboard/onboard.md]: can we throw in some links?

File: onboard/onboard.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - slack
+    - terraform
+    - aws-cli
```
Comment (zpatrick) [inline onboard/onboard.md]: [nit] probably add `(recommended)`

File: onboard/onboard.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    - l0
```
Comment (zpatrick) [inline onboard/onboard.md]: Can probably remove this one, since we expect users to build from the github repo

File: onboard/onboard.md (hunk 1 above)
Comment (nehayward) [reply onboard/onboard.md]: definitely I was thinking of making ansible or some script too.

File: onboard/onboard.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## Github Workflow
+
+1. Branch off of develop branch and name your branch with format issuenumber-brief-description.
+> git checkout -b "338-Refactor-API-Delete"
+
+2. After finishing code create a PR and add at least one reviewer and assign pr to yourself. Next, notify reviewer on Slack with DM or in #carbon_internal.
+> cc @zpatrick 
+
+3. Finally after approval, if there are no changes needed rebase and merge code, you can use CLI or [github.com](https://github.com/blog/2243-rebase-and-merge-pull-requests) to achieve this. This ensures a clean commit log.
```
Comment (zpatrick) [inline onboard/onboard.md]: this reads a little bit odd - maybe 
```
You may merge once all reviewers have approved your pull request. To ensure a clean commit log, use the [rebase](#) or [squash](#) option when merging.
```

Also, let's add a note on deleting the branch afterwords.

File: onboard/onboard.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    - git
+    - slack
+    - terraform
+    - aws-cli
+    - l0
+    - go
+
+## Github Workflow
+
+1. Branch off of develop branch and name your branch with format issuenumber-brief-description.
```
Comment (diemonster) [inline onboard/onboard.md]: This is mostly true, but I don't think we need to be so prescriptive. Generally speaking, we use Gitflow for our branching methodology. I suppose we could standardize on the names we use for feature branches, but I'm not sure if that gets us much at the moment.

File: onboard/onboard.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+2. After finishing code create a PR and add at least one reviewer and assign pr to yourself. Next, notify reviewer on Slack with DM or in #carbon_internal.
+> cc @zpatrick 
+
+3. Finally after approval, if there are no changes needed rebase and merge code, you can use CLI or [github.com](https://github.com/blog/2243-rebase-and-merge-pull-requests) to achieve this. This ensures a clean commit log.
+
+## Github Best Practices
+- If someone writes a comment on your PR and it starts with `[nit]` that just means "nitpick", and you don't have to change that if you don't want to
+
+## Slack
```
Comment (zpatrick) [inline onboard/onboard.md]: We can probably remove this part. Seems odd to have it in this public-facing doc since our slack channel is private.

**Comment by diemonster** [comment]
We also have this doc, which already addresses some of what's in here: https://github.com/quintilesims/layer0/blob/develop/.github/CONTRIBUTING.md

File: onboard/onboard.md (hunk 1 above)
Comment (diemonster) [reply onboard/onboard.md]: [nit] I think we should continue to avoid dependencies on config management tools ala Ansible, salt, etc.

**Comment by nehayward** [comment]
@diemonster I think a standard feature branch name would be helpful, this was just my best guess from looking at existing branches. Also I'll reference the contributing section in the GitHub section.

@zpatrick thanks for the feedback I'll go ahead and make those changes.

File: onboard/onboard.md (hunk 1 above)
Comment (nehayward) [reply onboard/onboard.md]: ^ I agree, I was more thinking for a basic script to install necessary tools for a fresh dev computer.

**Comment by zpatrick** [comment]
:+1: looks good to me

**Pull request merged**